src/python/epicscorelibs/path/cothread.py

```diff
@@ -18,8 +18,16 @@ def check_cothread_order():
     if "cothread" not in sys.modules:
         return
     cothread = sys.modules.get("cothread")
+
+    ver = ()
+    for c in cothread.__version__.split("."):
+        try:
+            ver = ver + (int(c), )
+        except ValueError:
+            # Ignore setuptools_scm auto-generated trailing parts of a version number e.g. "2.20.1.dev1+g82e2778.d20241105"
+            pass
+
     # >= 2.16 will attempt to import this module
-    ver = tuple(int(c) for c in cothread.__version__.split("."))
```

Comment:
Since the first two (or three?) pieces will always be integers, maybe some slicing would be simpler?

```py
ver = tuple(int(c) for c in cothread.__version__.split(".")[:2])
```

Reply:
That is indeed much cleaner. Implemented in [811a856](https://github.com/epics-base/epicscorelibs/pull/31/commits/811a8566b74dfd48d75c234e0689cc20b0529dea).